src/lib/operators/operator_join_predicate.cpp

```diff
@@ -53,4 +53,8 @@ void OperatorJoinPredicate::flip() {
   predicate_condition = flip_predicate_condition(predicate_condition);
 }
 
+bool operator<(const OperatorJoinPredicate& l, const OperatorJoinPredicate& r) {
+  return std::make_tuple(l.column_ids, l.predicate_condition) < std::make_tuple(r.column_ids, r.predicate_condition);
```

Comment:
Is the creation of a tuple really necessary here?

Reply:
less writing. It most probably won't be "created" at runtime

Reply:
I'd guess it's seldomly used? And just within the optimizer? 

If not, I would rather make sure that we don't create tuple unnecessarily.

Reply:
I'd bet a good chunk of... let's say a finger of my left hand that no extra machine-code is generated for this in release builds. And even if it were, this is not used in hot loops.